client/test_helpers: fix SetupLocalCosmosNode

cfal opened the pull request:
BCI-978

cfal committed:
client/test_helpers: terrad -> wasmd

cfal committed:
client/test_helpers: specify --home instead of --keyring-dir so that ~/.wasmd isn't created

cfal committed:
client/test_helpers: only listen locally

cfal committed:
remove terrad.nix, add wasmd.nix to install wasmd

archseer commented on wasmd.nix:
```diff
@@ -0,0 +1,28 @@
+    export GOSUMDB=off
+    export GOPROXY=direct
+    cd $src && make install
+    ln -s go/bin $out/bin
```
This should be under `installPhase`

archseer commented on wasmd.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "07nprps0da81krpl2a66qn823ybzghnx55zw7504wpynhgfa0bll";
+  };
+
+  sourceRoot = ".";
```
I think if you remove this then no need for `cd src`

cfal replied:
for some reason, i still needed "cd src", even without `sourceRoot`

cfal committed:
nix.conf: disable sandbox

cfal committed:
.github/workflows: set extra_nix_config

cfal committed:
fix derivation path, set ATOM coin type

cfal committed:
client/test_helpers: rpc.laddr parameter requires tcp as protocol

cfal committed:
client/test_helpers: fix hardcoded token

cfal committed:
cosmos/testdata: recompile my_first_contract.wasm

cfal committed:
client/test_helpers: fix out of gas error, set gas adjustment on deploy

cfal committed:
cosmos/client: fix wasm attribute filters when reading events

cfal committed:
client/client_test: fix wasm attribute check, remove count attribute check

.. because it doesn't exist in the updated contract reset function

https://github.com/CosmWasm/cw-template/blob/main/src/contract.rs#L69

cfal committed:
client/test_helpers: add required Label field when instantiating contract

cfal committed:
cosmos/client: register auth types with interface registry

cfal committed:
cosmos/client: don't hardcode chain id in DeployTestContract

cfal committed:
.github/workflows: cache nix
